### PR TITLE
[Merged by Bors] - Add readme as docs to relevant crates.

### LINKED ISSUE #2566: Include READMEs from Bevy crates as crate-level documentation
## How can Bevy's documentation be improved?

With the release of [Rust 1.54](https://blog.rust-lang.org/2021/07/29/Rust-1.54.0.html), we can now do this:

```rust
#![doc = include_str!("README.md")]
```

This is great, because we have useful content in our READMEs that should be in our module-level documentation, rather than buried several levels deep on Github.

Some crates already have existing module-level description; for the most part this should be consolidated into the readme. `bevy` itself is the only exception that comes to mind; it's probably worth including some advice on contributing / navigating Github that doesn't make sense on docs.rs.

### LINKED ISSUE #3005: Bevy ECS readme: not working example due to missing `#[derive(Component)]`
Hi everyone,

I'm tinkering with [`bevy_ecs`](https://github.com/bevyengine/bevy/tree/main/crates/bevy_ecs) and just realized that its README is not up-to-date with the [recent change](https://github.com/bevyengine/bevy/commit/07ed1d053e7946a116ce3eef273fc93dd246f49d) making `#[derive(Component)]` mandatory for all component structs.

Specifically
https://github.com/bevyengine/bevy/blob/6a8a8c9d21f32e0e46623db9438813b009f9e014/crates/bevy_ecs/README.md#L100-L130

Attempt to run the example would result in errors as below
```
the trait `bevy_ecs::component::Component` is not implemented for `Position`
the trait `bevy_ecs::component::Component` is not implemented for `Velocity`
```

Also, it could be worth it to mention that components are Rust structs but also need to implement bevy_ecs's `Component` trait. https://github.com/bevyengine/bevy/blob/6a8a8c9d21f32e0e46623db9438813b009f9e014/crates/bevy_ecs/README.md#L25-L29

### PR DESCRIPTION
**Pull request by Hoidigan**
Fixes #2566
Fixes #3005 

There are only READMEs in the 4 crates here (with the exception of bevy itself).
Those 4 crates are ecs, reflect, tasks, and transform.
These should each now include their respective README files.

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]
Awesome, thank you!

**Comment by alice-i-cecile** [comment]
bors try

**Comment by Nilirad** [comment]
I tried to do the same on my #2365 PR, but the examples in `README.md` break the doc test suite 😕

**Comment by alice-i-cecile** [comment]
@mockersf, any ideas on how to work around this CI issue?

**Comment by Nilirad** [comment]
It's not only a CI issue though. Running a simple command like

```shell
cargo test -p bevy_ecs --doc
```

would fail.

**Comment by Hoidigan** [comment]
We could ignore the doc tests? Put " ```rust ignore " for the start of the blocks instead of " ```rust "
Either that or go through and put the lines that get ignored but make the doc test work. I'm not sure if github would still show them or not, though.
(https://doc.rust-lang.org/rustdoc/documentation-tests.html#hiding-portions-of-the-example)

**Comment by alice-i-cecile** [comment]
>  I'm not sure if github would still show them or not, though.

Yeah, I think Github is likely to just show the extra lines of code :( Let's try ignoring them; it's no worse than the current situation where they're never compiled.

**Comment by Nilirad** [comment]
I don't know if ignoring will retain the code highlighting on GitHub renders...

**Comment by Hoidigan** [comment]
Hmmm, yeah I don't know either.
I'll try a few and see.

**Comment by alice-i-cecile** [comment]
https://github.com/bevyengine/bevy/blob/9f096ecb830b15943bef87d520e1c696de6868e7/crates/bevy_ecs/README.md works!

**Comment by Hoidigan** [comment]
I'll go through and grab the rest then 😄

**Comment by Hoidigan** [comment]
Alrighty they should be ignored now, cargo test --doc -p * worked fine for me.

**Comment by mockersf** [comment]
I'm not a fan of ignoring the failing code examples as a big plus of including the README is to test those examples. Would it be hard to fix them?

**Comment by Hoidigan** [comment]
The issue wasn't that they would be hard to fix (I don't think so but I'd need to check again once I can) but that if we fixed them we'd either clutter up examples in the docs and GitHub, or we'd do the hide lines thing for docs.rs but it would show the lines still on GitHub.

**Comment by mockersf** [comment]
> It's up a little bit in the conversation, we talked about it.

Yup, but it depends: is allowing the code blocks to work just adding 2 lines of imports? I think that's worth it. Is it adding 20 lines of setup? Ok that can be ignored, or maybe the example can be reworked?

**Comment by Hoidigan** [comment]
I'm not entirely sure and I'm not in a position to check right now, but in the next day or two I should be able to see. (Maybe tonight, I don't know.)

**Comment by Hoidigan** [comment]
Alrighty, so I went through and made the examples work. 
There were some things I was a little iffy about, namely declaring Position and Velocity for a lot of examples in bevy_ecs, but that fits in two lines. Most of them were solved by that and importing the prelude.
I did take bevy as a dev dependency to bring in the time resource.
Bevy_ecs does now have all working examples and none ignored. Most of them are only 3 or 4 lines longer. (and there was a few actual errors in them too, so that's fixed now.)

bevy_reflect has all of the setup at the top example. I *could* go through and copy and paste it into every other example but that's the 20 lines of setup we wanted to avoid. 
Would it be possible (or wanted) to create a small crate inside of reflect and then have a dev-dependency on that for some setup function?

Tasks and transform had no examples so that's easy enough.

**Comment by NiklasEi** [comment]
@Hoidigan could you add #3005 as getting resolved by this PR?

**Comment by Hoidigan** [comment]
Fixes #3005

**Comment by NiklasEi** [comment]
> Fixes #3005
> 
> (that is how you do that, right? Just put that here?)

I think it needs to be in the commit message. So in this case in the PR description.

File: crates/bevy_ecs/README.md [hunk 1]
```diff
@@ -68,18 +87,20 @@ fn print_position(query: Query<(Entity, &Position)>) {
 Apps often require unique resources, such as asset collections, renderers, audio servers, time, etc. Bevy ECS makes this pattern a first class citizen. `Resource` is a special kind of component that does not belong to any entity. Instead, it is identified uniquely by its type:
 
 ```rust
+use bevy_ecs::prelude::*;
+
 #[derive(Default)]
-struct Time {
-    seconds: f32,
```
Comment (cart) [inline crates/bevy_ecs/README.md:73]: I object to using a tuple struct here. The named field is way better at documenting intent. Tuple structs rarely have a place in public apis imo.

**Comment by cart** [comment]
One small comment. Everything else looks good to me!

File: crates/bevy_ecs/README.md (hunk 1 above)
Comment (Hoidigan) [reply crates/bevy_ecs/README.md:73]: Sounds good to me 👍

**Pull request closed without merging**